src/libraries/System.Collections/tests/Generic/Dictionary/Dictionary.Tests.cs

```diff
@@ -261,6 +261,41 @@ public void Remove_NonExistentEntries_DoesNotPreventEnumeration()
             }
         }
 
+        [Theory]
+        [InlineData(10)]
+        [InlineData(1000)]
+        [InlineData(1000_000)]
+        public void InsertionOpsOnly_Enumeration_PreservesInsertionOrder(int count)
+        {
+            var dictionary = new Dictionary<int, int>();
```

Comment:
I considered whether this should be part of the generic tests, but it didn't feel relevant to what we are testing.